[RM Ch07.9] IaaC Security Requirements (#2436)
cntt-n/CNTT: doc/ref_model/chapters/chapter07.md

<p align="center"><b>Table 7-8:</b> Compliance with standards requirements</p>

<a name="7.9.9"></a>
### 7.9.9. IaaC - Secure Design and Architecture Stage Requirements

| Ref | Requirement | Definition/Note |
|---|----|---|
| req.sec.arch.001 | Threat Modelling methodologies and tools **should** be used during the Secure Design and Architecture stage triggered by Software Feature Design trigger | Methodology to identify and understand threats impacting a resource or set of resources. It may be done manually or using tools like open source OWASP Threat Dragon |
| req.sec.arch.002 | Security Control Baseline Assessment **should** be performed during the Secure Design and Architecture stage triggered by Software Feature Design trigger | Typically done manually by internal or independent assessors.  |

<p align="center"><b>Table 7-9:</b> IaaC - Secure Design and Architecture Stage Requirements</p>

<a name="7.9.10"></a>
### 7.9.10. IaaC - Secure Code Stage Requirements

| Ref | Requirement | Definition/Note |
|---|----|---|

> | req.sec.code.001 | SAST -Static Application Security Testing **should** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code for software vulnerabilities and gaps against best practices. Example: open source OWASP range of tools.|

Reviewer: ```suggestion
| req.sec.code.001 | SAST -Static Application Security Testing **must** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code for software vulnerabilities and gaps against best practices. Example: open source OWASP range of tools.|
```

| req.sec.code.002 | SCA – Software Composition Analysis **should** be applied during Secure Coding stage triggered by Pull, Clone or Comment trigger. | Security testing that analyses application source code or compiled code for software components with known vulnerabilities. Example: open source OWASP range of tools.  |
| req.sec.code.003 | Source Code Review **should** be performed continuously during Secure Coding stage. | Typically done manually.  |

> | req.sec.code.004 | Integrated SAST via IDE Plugins **should** be used during Secure Coding stage triggered by Developer Code trigger. | On the local machine: through the IDE or integrated test suites; triggered on completion of coding be developer. |

Reviewer: This is an implementation of req.sec.code.001 -- does it belong in RM?

> | req.sec.code.005 | ISAST of Source Code Repo **should** be used during Secure Coding stage triggered by Developer Code trigger. | Continuous delivery pre-deployment: scanning prior to deployment. |

Reviewer: Is this requirement about Static testing of the repo? If yes, then suggest
```suggestion
| req.sec.code.005 | SAST of Source Code Repo **should** be performed during Secure Coding stage triggered by Developer Code trigger. | Continuous delivery pre-deployment: scanning prior to deployment. |
```

<p align="center"><b>Table 7-10:</b> IaaC - Secure Code Stage Requirements</p>

<a name="7.9.11"></a>
### 7.9.11. IaaC - Continuous Build, Integration and Testing Stage Requirements

| Ref | Requirement | Definition/Note |
|---|----|---|
| req.sec.bld.001 | SAST -Static Application Security Testing **should** be applied during the Continuous Build, Integration and Testing stage triggered by Build and Integrate trigger. | Example: open source OWASP range of tools.|
| req.sec.bld.002 | SCA – Software Composition Analysis **should** be applied during the Continuous Build, Integration and Testing stage triggered by Build and Integrate trigger. | Example: open source OWASP range of tools.  |
| req.sec.bld.003 | Container and Image Scan **must** be applied during the Continuous Build, Integration and Testing stage triggered by Package trigger. | Example: A push of a container image to a container registry may trigger a vulnerability scan before the image becomes available in the registry.  |

> | req.sec.bld.004 | DAST – Dynamic Application Security Testing **must** be applied during the Continuous Build, Integration and Testing stage triggered by Stage & Test trigger. | Security testing that analyses a running application by exercising application functionality and detecting vulnerabilities based on application behaviour and response. Example: OWASP ZAP. |

Reviewer: ```suggestion
| req.sec.bld.004 | DAST – Dynamic Application Security Testing **should** be applied during the Continuous Build, Integration and Testing stage triggered by Stage & Test trigger. | Security testing that analyses a running application by exercising application functionality and detecting vulnerabilities based on application behaviour and response. Example: OWASP ZAP. |
```
